### PR TITLE
Add Prometheus metrics.

### PR DESCRIPTION
**Pull request by SuperQ**
* Add Prometheus metrics handler.
* Add build info metric.
* Update Makefile to add version metric info.

### CODE REVIEW

File: software/earl/Makefile [hunk 1]
```diff
@@ -2,14 +2,17 @@
...
-VERSION ?= $(shell git log --date=short --pretty=format:'%h@%cd' -n 1 .)
+BRANCH    ?= $(shell git rev-parse --abbrev-ref HEAD)
+BUILDDATE ?= $(shell date --iso-8601=seconds)
+REVISION  ?= $(shell git rev-parse HEAD)
+VERSION   ?= $(shell git log --date=short --pretty=format:'%h@%cd' -n 1 .)
 
 all : earl test
 
-# Note, our go version on the Pi does not understand the newer main.VERSION=xxx
+# Note, our go version on the Pi does not understand the newer main.Version=xxx
```
Comment (hzeller) [inline software/earl/Makefile]: So, this comment can be removed now as we now build with a newer go for the Pi.

**Review by hzeller — CHANGES_REQUESTED** [review]
LGTM

**Comment by SuperQ** [comment]
Fixed Makefile comment.

**Review by hzeller — APPROVED** [review]
LGTM

**Pull request merged**